velox/exec/tests/AddressableNonNullValueListTest.cpp

```diff
@@ -75,6 +98,52 @@ class AddressableNonNullValueListTest : public testing::Test,
     test::assertEqualVectors(uniqueData, copy);
   }
 
+  void testSerialization(
+      const std::vector<T>& entries,
+      vector_size_t totalSize,
+      const VectorPtr& uniqueData) {
+    // Test copy/appendSerialized round-trip for AddressableNonNullValueList.
+    // Steps in the test:
+    // i) Copy entry length, hash and value of each entry to a stream.
+    // ii) Deserialize stream to a new set of entries.
+    // iii) Read deserialized entries back into a vector.
+    // iv) Validate the result vector.
+    size_t offset = 0;
+    auto* rawBuffer =
+        AlignedBuffer::allocate<char>(totalSize, pool())->as<char>();
```

Comment:
This pattern is problematic. The buffer created by allocate() is a temp variable that gets destroyed after this statement leaving rawBuffer as a dangling pointer.

Reply:
Interesting... It now makes sense why all other code creates a variable for the allocation and then type-casts it. Revised this code.